apps/dot-cdn/src/app/dotcdn.service.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable } from '@angular/core';
+import { ReplaySubject } from 'rxjs';
+import { CoreWebService, SiteService } from '@dotcms/dotcms-js';
+import { pluck, mergeMap } from 'rxjs/operators';
+import * as moment from 'moment';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class DotCDNService {
+    currentSite$: ReplaySubject<string> = new ReplaySubject(null);
+
+    constructor(private coreWebService: CoreWebService, private siteService: SiteService) {
+        this.siteService
+            .getCurrentSite()
+            .pipe(pluck('identifier'))
+            .subscribe((hostId: string) => {
+                this.currentSite$.next(hostId);
+            });
+    }
+
+    /**
+     * Request stats via Core Web Service
+     *
+     * @param {string} period
+     * @return {*}  {*} // TODO: Find appropriate return type
+     * @memberof DotCDNService
+     */
+    requestStats(period: string): any {
+        return this.currentSite$.pipe(
+            mergeMap((hostId: string) => {
+                const dateTo = moment().format('YYYY-MM-DD');
+                const dateFrom = moment().subtract(period, 'd').format('YYYY-MM-DD');
+                return this.coreWebService.request({
+                    url: `/api/v1/dotcdn/stats?hostId=${hostId}&dateFrom=${dateFrom}&dateTo=${dateTo}`
+                });
+            })
+        );
+    }
+
+    /**
+     *  Makes a request to purge the cache
+     *
+     * @param {string[]} [urls=[]]
+     * @param {boolean} [invalidateAll=false]
+     * @return {*}
+     * @memberof DotCDNService
+     */
+    purgeCache(urls: string[] = [], invalidateAll: boolean = false) {
+        return this.currentSite$.pipe(
```

Comment:
I think you are doing an extra request here to get the site id but you already have it in a static property in the site service. Please double check.

Reply:
@fmontes Nope. I'm musing ReplaySubject which emits old values. A new currentSite is provided on page reload.